src/test/app/LedgerLoad_test.cpp

```diff
@@ -0,0 +1,229 @@
+        {
+            Env env(*this, ledgerConfig("", Config::LOAD_FILE));
+        });
+
+        // file does not exist
+        except ([this]
+        {
+            Env env(*this, ledgerConfig("badfile.json", Config::LOAD_FILE));
+        });
+
+        // make a corrupted version of the ledger file (last 10 bytes removed).
+        boost::system::error_code ec;
+        auto ledgerFileCorrupt = dbPath_ / "ledgerdata_bad.json";
+        copy_file(
+            ledgerFile_,
+            ledgerFileCorrupt,
+            copy_option::overwrite_if_exists,
+            ec);
+        if(! BEAST_EXPECTS(!ec, ec.message()))
+            return;
+        auto filesize = file_size(ledgerFileCorrupt, ec);
+        if(! BEAST_EXPECTS(!ec, ec.message()))
+            return;
+        resize_file(ledgerFileCorrupt, filesize - 10, ec);
+        if(! BEAST_EXPECTS(!ec, ec.message()))
+            return;
+
+        except ([this, &ledgerFileCorrupt]
+        {
+            Env env(*this, ledgerConfig(ledgerFileCorrupt.string(), Config::LOAD_FILE));
+        });
+    }
+
+    void
+    testLoadByHash ()
+    {
+        testcase ("Load by hash");
+        using namespace test::jtx;
+
+        // create a new env with the ledger hash specified for startup
+        auto ledgerHash = to_string(jvHashes_[jvHashes_.size()-1]);
+        boost::erase_all(ledgerHash, "\"");
+        Env env(*this, ledgerConfig(ledgerHash, Config::LOAD));
+        auto jrb = env.rpc ( "ledger", "current", "full") [jss::result];
+        BEAST_EXPECT(jrb[jss::ledger][jss::accountState].size() == 97);
+        BEAST_EXPECT(
+            jrb[jss::ledger][jss::accountState].size() <=
+            jvLedger_[jss::ledger][jss::accountState].size());
+    }
+
+    void
+    testLoadLatest ()
+    {
+        testcase ("Load by keyword");
+        using namespace test::jtx;
+
+        // create a new env with the ledger "latest" specified for startup
+        Env env(*this, ledgerConfig("latest", Config::LOAD));
+        auto jrb = env.rpc ( "ledger", "current", "full") [jss::result];
+        BEAST_EXPECT(
+            jvLedger_[jss::ledger][jss::accountState].size() ==
+            jrb[jss::ledger][jss::accountState].size());
+    }
+
+    void
+    testLoadIndex ()
+    {
+        testcase ("Load by index");
+        using namespace test::jtx;
+
+        // create a new env with specific ledger index at startup
+        Env env(*this, ledgerConfig("43", Config::LOAD));
+        auto jrb = env.rpc ( "ledger", "current", "full") [jss::result];
+        BEAST_EXPECT(
+            jvLedger_[jss::ledger][jss::accountState].size() ==
+            jrb[jss::ledger][jss::accountState].size());
+    }
+
+public:
+    void run ()
+    {
+        setupLedger();
+        BOOST_SCOPE_EXIT( this_ ) {
+            boost::system::error_code ec;
+            boost::filesystem::remove_all(this_->dbPath_, ec);
+            this_->expect(!ec, ec.message(), __FILE__, __LINE__);
+        } BOOST_SCOPE_EXIT_END
```

Comment:
It's a little frustrating that `make_unique` is not available to help with this.  We could write a little `make_scope` factory function around this idiom to play the role of `make_unique`.  That's probably not worth it just for this one instance.  But if we find ourselves doing this more places in the future, then it would be worth it.